numpy/core/src/multiarray/item_selection.c

```diff
@@ -902,7 +903,7 @@ PyArray_Choose(PyArrayObject *ip, PyObject *op, PyArrayObject *out,
         Py_XDECREF(mps[i]);
     }
     Py_DECREF(ap);
-    npy_free_cache(mps, n * sizeof(mps[0]));
+    PyDataMem_FREE(mps);
```

Comment:
This (and the one below) will slightly increase the cache misses, but is more consistent with the PyDataMem_NEW allocation